package/MDAnalysis/coordinates/XYZ.py

```diff
@@ -367,8 +367,11 @@ def _read_next_timestep(self, ts=None):
             # we assume that there are only two header lines per frame
             f.readline()
             f.readline()
+            # convert all entries at the end once for optimal speed
+            tmp_buf = []
             for i in range(self.n_atoms):
```

Comment:
Out of scope for this PR, but I'm also going to look into if it's faster to iterate on the file rather than repeatedly call `readline`